src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,264 @@
+﻿using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Newtonsoft.Json.Linq;
+using SonarQube.Client.Helpers;
+using SonarQube.Client.Models;
+
+namespace SonarQube.Client.Services
+{
+    public class SonarQubeClient : ISonarQubeClient
+    {
+        private readonly HttpMessageHandler messageHandler;
+        private readonly TimeSpan requestTimeout;
+
+        public SonarQubeClient(HttpMessageHandler messageHandler, TimeSpan requestTimeout)
+        {
+            this.messageHandler = messageHandler;
+            this.requestTimeout = requestTimeout;
+        }
+
+        public Task<Result<ComponentDTO[]>> GetComponentsSearchProjectsAsync(ConnectionDTO connection,
+            ComponentRequest request, CancellationToken token)
+        {
+            const string SearchProjectsAPI = "api/components/search_projects"; // Since 6.2; internal
```

Comment:
SQService already knows the version, yet if it is lower than one in these comments, client still makes the call. Perhaps pass the version, add an attribute "MinimalSupportedVersion" and then make the check before making the call.

Reply:
Fine with that but what different behavior than the `HttpStatusCode.NotFound` do you want to provide?

Reply:
I'd expect result to be "not supported by server" and leave it for SQService to handle.